### DISCUSSION TITLE
Geo add the ability to get a reference to the map instance

### DISCUSSION THREAD

**Discussion by samouss** [algolia/instantsearch#5215] [Ideas]
**Is your feature request related to a problem? Please describe 🙏**

It might happen in some cases that you directly want to interact with the map instance. Right now the only way to get this instance is through the event listeners on the markers. It would be nice to provide it as soon as the map is instantiated.

**Describe the solution you'd like 🤔**

We can trigger an event once the map is instantiated.

```js
search.addWidget(
  instantsearch.widgets.geoSearch({
    container: "#geo-search-container",
    googleReference: window.google,
    events: {
      onMapCreated: instance => {
        // Do something with the instance
      }
    }
  })
);
```

**Additional context**

See https://github.com/algolia/instantsearch.js/issues/2778#issuecomment-411552188

**Comment by Haroenv** [comment]
another option would be to add this event to the return of `.geoSearch()`, but that would be less obvious to deal with, so adding `events` here seems nice. Maybe it can be added to other ones too if it makes sense

**Comment by bobylito** [comment]
Have you considered having the the instance of the map as a property of the widget and documenting it?

**Comment by Haroenv** [comment]
@bobylito then it would be undefined before mounted right?

**Comment by bobylito** [comment]
> @bobylito then it would be undefined before mounted right?

Probably indeed 😅 . Other idea: what about having a method on the widget instance that returns a promise of the map?

**Comment by samo012** [comment]
For anyone still wondering, a reference to the map is passed in the events from [builtInMarker](https://www.algolia.com/doc/api-reference/widgets/geo-search/js/#widget-param-builtinmarker) and [customHTMLMarker](https://www.algolia.com/doc/api-reference/widgets/geo-search/js/#widget-param-customhtmlmarker).
